src/codeql.ts

```diff
@@ -654,6 +663,19 @@ function getCodeQLForCmd(cmd: string): CodeQL {
         errorMatchers
       );
     },
+    async resolveLanguages() {
+      const codeqlArgs = ["resolve", "languages", "--format=json"];
+      let output = "";
+      await new toolrunner.ToolRunner(cmd, codeqlArgs, {
+        listeners: {
+          stdout: (data: Buffer) => {
+            output += data.toString();
+          },
+        },
+      }).exec();
+
+      return JSON.parse(output);
```

Comment:
Catch malformed json errors and rethrow with a meaningful message.

```suggestion
      try {
        return JSON.parse(output);
      } catch (e) {
        throw new Error("Something meaningful.);
      }
```

Reply:
Wouldn't that suppress the likely interesting error message of `e` ? In Java I'd write `new Error("Unexpected output from codeql resolve languages", e)` . What would be the equivalent in Javascript?

Reply:
I think just construct the message as a string. So like `new Error("Unexpected output from codeql resolve languages: " + e.message)` and maybe some type checks to make sure that `e.message` exists.

Reply:
I opted to just write `Unexpected output from codeql resolve languages: ${e}` . This should include the original error's type and side-steps the problem of a potentially empty message string.